src/lib/logger.ts

```diff
@@ -0,0 +1,364 @@
+			return 'vip';
+		}
+
+		// Extract the command name (first non-flag argument)
+		let cmdName = 'vip';
+		for ( const arg of args ) {
+			if ( ! arg.startsWith( '-' ) ) {
+				cmdName = arg;
+				break;
+			}
+		}
+
+		// If the process is running a specific bin file like vip-dev-env-start.js,
+		// extract the command name from the filename
+		const processPath = process.argv[ 1 ] || '';
+		const match = processPath.match( /vip-([\w-]+)\.js$/ );
+		if ( match ) {
+			cmdName = match[ 1 ];
+		}
+
+		return cmdName;
+	}
+
+	private _createLogFilePath(): string {
+		// Create a timestamp in a readable format: YYYY-MM-DD_HH-MM-SS
+		const now = new Date();
+		const timestamp = now
+			.toISOString()
+			.replace( /:/g, '-' )
+			.replace( /\..+/, '' )
+			.replace( 'T', '_' );
+
+		// Create a filename with command name and timestamp
+		const filename = `${ this.commandName }_${ timestamp }.log`;
+
+		return path.join( this.logDir, filename );
+	}
+
+	private _parseDebugEnv( debugEnv: string ): Set< string > {
+		const namespaces = new Set< string >();
+		const patterns = debugEnv.split( /[\s,]+/ );
+
+		for ( const pattern of patterns ) {
+			if ( ! pattern ) continue;
+
+			// Handle negated patterns
+			const isExclude = pattern.startsWith( '-' );
+			const ns = isExclude ? pattern.slice( 1 ) : pattern;
+
+			if ( isExclude ) {
+				namespaces.delete( ns );
+			} else {
+				namespaces.add( ns );
+			}
+		}
+
+		return namespaces;
+	}
+
+	private _createRootLogger( options: VIPLoggerOptions ): winston.Logger {
+		// Determine console log level based on debug mode
+		const consoleLevel = options.consoleLevel || ( this.isDebugMode ? 'debug' : 'warn' );
+		const fileLevel = options.fileLevel || 'debug'; // Always log everything to file
+
+		return winston.createLogger( {
+			level: 'debug', // Set internal level to debug to capture everything
+			format: winston.format.combine( winston.format.timestamp(), winston.format.json() ),
+			transports: [
+				new winston.transports.Console( {
+					level: consoleLevel,
+					format: winston.format.combine(
+						winston.format.colorize(),
+						winston.format.timestamp( { format: 'HH:mm:ss' } ),
+						winston.format.printf(
+							( { timestamp, level, message, namespace = 'app' } ) =>
+								`${ timestamp } ${ namespace } ${ level }: ${ message }`
+						)
+					),
+				} ),
+				new winston.transports.File( {
+					level: fileLevel,
+					filename: this.logFilePath,
+					format: winston.format.combine(
+						winston.format.timestamp(),
+						winston.format.printf(
+							( { timestamp, level, message, namespace = 'app' } ) =>
+								`${ timestamp } ${ namespace } ${ level }: ${ message }`
+						)
+					),
+				} ),
+			],
+		} );
+	}
+
+	// Check if a namespace is enabled (compatible with debug.enabled())
+	public isEnabled( namespace: string ): boolean {
+		// If '*' is enabled, everything is enabled
+		if ( this.enabledNamespaces.has( '*' ) ) {
+			return true;
+		}
```

Comment:
Will this work the same as with `DEBUG=*,-modem`? That is, allow all namespaces except `modem`?